src/mappings/entities/mapIdentities.ts

```diff
@@ -361,3 +369,39 @@ const handleSecondaryKeysAdded = async (
 
   await Promise.all(promises);
 };
+
+const handlePrimaryKeyUpdated = async (
+  blockId: string,
+  eventId,
+  params: Codec[]
+): Promise<void> => {
+  const [rawDid, , newKey] = params;
+  const address = getTextValue(newKey);
+
+  const identity = await getIdentity(getTextValue(rawDid));
+
+  Object.assign(identity, {
+    primaryAccount: address,
+    updatedBlockId: blockId,
+    eventId,
+  });
+
+  await identity.save();
+};
+
+const handleSecondaryKeyLeftIdentity = async (params: Codec[]): Promise<void> => {
+  const [, rawAccount] = params;
+
+  const account = rawAccount.toString() as MeshAccount;
+
+  let address;
+  if (typeof account === 'string') {
+    // for chain version >= 5.0.0
+    address = account;
+  } else {
+    // for chain version < 5.0.0
+    ({ account: address } = account);
```

Comment:
`account` is only made from a `.toString` call, so any destructure will be a run time error.

I think we do the `.toString`, then feed that to `JSON.parse` for this to work